src/protocol/libp2p/kademlia/query/get_record.rs

```diff
@@ -36,22 +36,55 @@ use std::collections::{BTreeMap, HashMap, HashSet, VecDeque};
 /// Logging target for the file.
 const LOG_TARGET: &str = "litep2p::ipfs::kademlia::query::get_record";
 
+/// The configuration needed to instantiate a new [`GetRecordContext`].
 #[derive(Debug)]
-pub struct GetRecordContext {
+pub struct GetRecordConfig {
     /// Local peer ID.
-    local_peer_id: PeerId,
+    pub local_peer_id: PeerId,
 
-    /// How many records have been successfully found.
+    /// How many records we already know about (ie extracted from storage).
     pub record_count: usize,
```

Comment:
May be rename this to `known_records`? And add a comment that this can be either `0` or `1`.